### DISCUSSION TITLE
[API Request] Add song.downloadSize to better reflect the download size, not the size on disk

### DISCUSSION THREAD

**Discussion by dbrader** [opensubsonic/open-subsonic-api#34] [API requests]
### Type of change

API tweak

### Proposal description

Add a song.downloadSize field to better reflect the amount of data the client should have downloaded.

Currently, song.size is the size the song takes up on disk, which is perfectly fine for mp3's in your library.  But when you have a flac or other format that gets transcoded before download, the client is unable to determine if the entire file was downloaded correctly.

### Backward compatibility impact

Adding a new attribute should not break any client.

### Backward compatibility

- [X] No backward compatibility impact.

### API details

Add a new downloadSize attribute to song entity.

```
<song id="34126" parent="34114" isDir="false" title="Africa" album="Toto IV" artist="Toto" track="10" year="1982" genre="AOR Classic Rock" coverArt="34114" size="5291484" contentType="audio/mpeg" suffix="mp3" duration="263" bitRate="160" path="Toto/Toto IV/10 Africa.mp3" playCount="35" discNumber="1" created="2012-02-15T06:19:39.000Z" starred="2023-01-17T04:00:42.176Z" albumId="2881" artistId="602" type="music" downloadSize="5291484" />
```	

### Security impacts

none

### Alternative solutions

You can only determine corrupt locally cached mp3 song files.

**Comment by deluan** [comment]
This is not possible if the server transcodes the files on the fly (which I assume most servers do), so it will not have the transcoded size before hand.

**Reply by lachlan-00** [reply]
yeah i can't predict that very well. i've actually taken off the estimates from the content-length as it was killing streams early

**Reply by Tolriq** [reply]
Exactly the reason for https://github.com/opensubsonic/open-subsonic-api/discussions/21 :)
